server/src/controllers/address.ts

```diff
@@ -0,0 +1,34 @@
+import { NextFunction, Request, Response } from "express";
+import { getAllCountriesService, getStatesForService, getCitiesForService, getAllAddressOfUserService, addNewAddressService } from "../services/address";
+import { countryModel, stateModel, cityModel } from "../models/types";
+
+export const getAllCountries = async (req: Request, res: Response) => {
+    const data: countryModel[] = await getAllCountriesService();
+    res.send(data);
+}
+
+export const addNewAddress = async (req: Request, res: Response) => {
+    const { userEmail, address, city, state, zipCode, country } = req.body;
+    const user_email = userEmail;
+    const zip_code = zipCode;
+    const data: number = await addNewAddressService(user_email, address, city, state, zip_code, country);
+    return res.send(data.toString());
+}
+
+export const getStatesFor = async (req: Request, res: Response) => {
+    const country_id = Number(req.query.countryId as string);
```

Comment:
Can add + behind req.query.countryId, and why are you typecasting it as string first and then converting to number ?

Reply:
what do you mean by adding + behind?

Reply:
Reference: 

const x = +"1000";
This will change "1000" to 1000